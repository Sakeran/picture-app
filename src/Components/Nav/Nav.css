.Nav {

}

.Nav ul {
  list-style: none;
  display: flex;
  margin-right: 0.3em;
}

.Nav ul li {
  font-weight: bold;
  font-size: 0.85em;
  margin-left: 1rem;
}
