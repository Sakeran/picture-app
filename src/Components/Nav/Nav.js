import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './Nav.css';

const Nav = (props) => (
  <nav className={"Nav" + (props.toggled ? " Nav-is-toggled" : "") }>
    {
      props.user ?
        <ul>
          <li><Link to="/logout">Log Out</Link></li>
        </ul>
      :
        <ul>
          <li><Link to="/login">Log In</Link></li>
          <li><Link to="/signup">Sign Up</Link></li>
        </ul>
    }
  </nav>
);

Nav.propTypes = {
  user: PropTypes.object,
  toggled: PropTypes.bool
};

export default Nav;
