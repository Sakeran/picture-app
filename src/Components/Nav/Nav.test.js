import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Nav from './Nav';
import { Link } from 'react-router-dom';

test('Nav component renders', () => {
  const component = shallow(<Nav />);
});

test('Nav component shows login link when logged out', () => {
  const user = null;
  const component = shallow(<Nav user={user} />);
  expect(component.find({to: "/login"}).exists()).toBe(true);
  expect(component.find('#logoutBtn').exists()).toBe(false);
  const tree = toJSON(component);
  expect(tree).toMatchSnapshot();
});

test('Nav component shows logout link when logged in', () => {
  const user = {data : 'Some User Data'};
  const component = shallow(<Nav user={user} />);
  expect(component.find('#logoutBtn').exists()).toBe(true);
  expect(component.find({to: "/login"}).exists()).toBe(false);
  const tree = toJSON(component);
  expect(tree).toMatchSnapshot();
});

test('Nav component has proper class when menu toggled on', () => {
  const component = shallow(<Nav toggled={true} />);
  expect(component.find('.Nav-is-toggled').exists()).toBe(true);
});

test('Nav component has proper class when menu toggled off', () => {
  const component = shallow(<Nav toggled={false} />);
  expect(component.find('.Nav-is-toggled').exists()).toBe(false);
});
