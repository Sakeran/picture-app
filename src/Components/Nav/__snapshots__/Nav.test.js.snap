// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component shows login link when logged out 1`] = `
<nav
  className="Nav"
>
  <ul>
    <li>
      <Link
        replace={false}
        to="/login"
      >
        Log In
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/signup"
      >
        Sign Up
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`Nav component shows logout link when logged in 1`] = `
<nav
  className="Nav"
>
  <ul>
    <li>
      <button
        id="logoutBtn"
      >
        Log Out
      </button>
    </li>
  </ul>
</nav>
`;
